josharagon/Refactor-Tractor
# Feature/post user data

## Pull Request

<br>

### Description

* Is this a feature or a fix?
Feature
- This creates posting functionality for the user. The POSTs are all done in one function in a new file. 
- Bug fix: Adjusts when today and user ID is created so that
-  a. they are stored as global variables
-  b. Today is redefines as the latest date that the current user has in their data 
- Reorganized some of our functions into helper functions for SRP
- Added in conditional that checks if a current user exists before reassigning one which allows us to reuse the fetchCurrentData functionality after our post data 

* Where should the reviewer start?
Review the PostData.js file to see the posting capabilities. 
Review the postData functionality in scripts.js

*Known bug
Data is currently being displayed multiple times. This can be easily fixed when we connect this functionality to the DOM display.
We do not yet have place to send in values to post so current posted data is hard coded in.  

* What functionalities do these changes effect? Why were these changes made? What was the motivation behind these changes?
This allows user to post new data as per the requirements of the app. It allows us to POST and refetch data.

* What outside features or research did you use to implement this fix?
*n/a

* How should this be tested?
Posting is currently connected to hard coded data on keydown. Once the page is loaded click down any key to see new posted data. 

<br>

***

#### Please review considerations below:
